File: twoSums/c/twoSum.c
#include <stdlib.h>
#include <stdio.h>

/*
* Note: The returned array must be malloced, assume caller calls free().
*/

int* twoSum(int* nums, int numsSize, int target) {
	int* p;
	p = nums;
	int i, j;

	int *res = NULL;
	res = (int *)malloc(2*sizeof(int));
	for (i = 0; i < numsSize; i++) {
		for(j = i+1; j < numsSize; j++) {
			if ((*(p + i) + *(p + j)) == target) {
				*res = i;
				*(res+1) = j;
				return res;
			}
		}
	}

    return NULL;

}

int main() {
	int size = 10;
	int *tests = NULL;
	tests = (int *)malloc(size*sizeof(int));
	int i = 0;
	for (i = 0; i < size; i++) {
		*(tests+i) = i;
	}

	int *p;
	p = twoSum(tests, size, 3);

    if (p) {
		for (i = 0; i < 2; i++) {
			printf("%d ", *(p+i));
		}
    } else {
		printf("p is NULL\n");
	}

    free(tests);
    free(p);
	return 0;
}

File: twoSums/go/src/main/twoSum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	var tMap map[int]int
	tMap = make(map[int]int)
	arr := make([]int, 2)
	for i := 0; i < len(nums); i++ {
		tMap[nums[i]] = i
	}

	for i := 0; i < len(nums); i++ {
		complement := target - nums[i]
		value, ok := tMap[complement]
		if ok && (value != i) {
			arr[0] = i
			arr[1] = value
			break
		}
	}
	return arr

}

func main() {
	res := make([]int, 2)
	nums := [4]int{2, 7, 11, 15}
	target := 9
	var numslice []int = nums[:]
	res = twoSum(numslice, target)
	for i := 0; i < len(res); i++ {
		fmt.Println(res[i])
	}

}
